scripts/install.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+#
+#  Copyright (c) 2024 Red Hat, Inc.
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  You may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+# Script to handle CLI helm installation for OCP
+#
+# Requires: oc, helm and an active login session to a cluster.
+
+set -e
+
+usage() {
+    echo "
+This script simplifies and automates the installation process of Helm charts on the OpenShift Container Platform (OCP) clusters.
+It ensures that the user is logged into a cluster and attempts to detect the cluster router base, updating the Helm chart configuration accordingly.
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --router-base <router-base> : Manually provide the cluster router base if auto-detection fails.
+  --release-name <name>       : Specify a custom release name for the Helm chart.
+  --generate-name             : Generate a name for the Helm release (overrides --release-name).
+  --namespace <namespace>     : Specify the namespace for the Helm release (autodetects if not provided).
+  --values <file>             : Specify your own values file for the Helm chart.
+  --help                      : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+  $0 --release-name myrelease      # Installs the Helm chart with the specified release name
+  $0 --generate-name               # Generates a name for the Helm release
+  $0 --values /path/to/values.yaml # Installs the Helm chart using the specified values file
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check for required commands
+if ! command_exists helm; then
+    echo "Error: 'helm' is required. Install it from https://docs.openshift.com/container-platform/4.16/applications/working_with_helm_charts/installing-helm.html to continue."
+    exit 1
+fi
+if ! command_exists oc; then
+    echo "Error: 'oc' is required. Install it from https://docs.openshift.com/container-platform/4.16/cli_reference/openshift_cli/getting-started-cli.html to continue."
+    exit 1
+fi
+
+# Check if required files and directories exist
+HELM_CHART_DIR="$(dirname "$0")/../charts/backstage"
```

Comment:
this requires that this repo be cloned. Is there a way to achieve the same thing from a single command, like https://github.com/rhdh-bot/openshift-helm-charts/blob/rhdh-1-rhel-9/installation/install.sh ?